### ISSUE TITLE
`checkConditionals` in `no-immediate-mutation` seems to be `true` by default, but mentioned as `false` in docs

### ISSUE THREAD

**Issue by krislintigo** [sindresorhus/eslint-plugin-unicorn#3722]
I have recommended setup in my project, meaning that ESLint is using default configuration of `no-immediate-mutation` rule. However, I'm observing error when seeing code pattern below:

```ts
const $and: Array<Record<string, unknown>> = []

  if (query.search) { // ESLint: Immediate mutation on array is not allowed. (unicorn/no-immediate-mutation)
    $and.push(getSearchQueryPart(search))
  }

  if (query.selectedRatings.length > 0) {
    $and.push(getSelectedRatingsQueryPart(query.selectedRatings))
  }
```

According to documentation, this should not be reported, and I support this default for the rule because it makes the code easier to read.

**Comment by krislintigo** [comment]
My bad, I was reading the documentation going ahead of release features. There is no `checkConditionals` option yet

<img width="663" height="150" alt="Image" src="https://github.com/user-attachments/assets/da98d756-9232-4c3d-b8f3-69babb383f1f" />
